src/api/api_client.rs

```diff
@@ -102,50 +107,56 @@ use std::convert::{TryFrom, TryInto};
 /// }
 ///
 /// let client = MyClient::new();
-/// let _api = Api::<(), _, PlainTipExtrinsicParams>::new(client);
+/// let _api = Api::<(), _, PlainTipExtrinsicParams<Runtime>, Runtime>::new(client);
 ///
 /// ```
 #[derive(Clone)]
-pub struct Api<P, Client, Params>
+pub struct Api<Signer, Client, Params, Runtime>
 where
 	Client: RpcClient,
-	Params: ExtrinsicParams<Index, Hash>,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: frame_system::Config + pallet_balances::Config,
+	Runtime::Hash: FromHexString,
+	Runtime::Balance: TryFrom<NumberOrHex>,
 {
-	signer: Option<P>,
-	genesis_hash: Hash,
+	signer: Option<Signer>,
+	genesis_hash: Runtime::Hash,
 	metadata: Metadata,
 	runtime_version: RuntimeVersion,
 	client: Client,
 	extrinsic_params_builder: Option<Params::OtherParams>,
 }
 
-/// Setter and getter calls to the local cache, no substrate node calls involved.
-impl<P, Client, Params> Api<P, Client, Params>
+impl<Signer, Client, Params, Runtime> Api<Signer, Client, Params, Runtime>
 where
-	P: Pair,
-	MultiSigner: From<P::Public>,
+	Signer: Pair,
+	MultiSigner: From<Signer::Public>,
 	Client: RpcClient,
-	Params: ExtrinsicParams<Index, Hash>,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: frame_system::Config + pallet_balances::Config,
+	Runtime::Hash: FromHexString,
+	Runtime::Index: From<u32>,
+	Runtime::Balance: TryFrom<NumberOrHex> + FromStr,
 {
 	/// Set the api signer account.
-	pub fn set_signer(&mut self, signer: P) {
+	pub fn set_signer(&mut self, signer: Signer) {
 		self.signer = Some(signer);
 	}
 
 	/// Get the public part of the api signer account.
-	pub fn signer_account(&self) -> Option<AccountId> {
+	pub fn signer_account(&self) -> Option<AccountId32> {
 		let pair = self.signer.as_ref()?;
 		let multi_signer = MultiSigner::from(pair.public());
 		Some(multi_signer.into_account())
```

Comment:
MultiSigner only implements `into_account` for AccountId32. Not sure why, but I guess there's a reason.

Reply:
Yeah, because moonbeam does in fact implement an account id with 20 bytes, but but still use the multisigner if I am not mistaken.